syntax = "proto3";

package auth;

// Сообщение запроса для аутентификации и регистрации
message AuthRequest {
  string username = 1;
  string password = 2;
}

// Сообщение ответа для аутентификации и регистрации
message AuthResponse {
  bool authenticated = 1;
  string message = 2;
  string token = 3; // например, токен сессии, пока может быть именем пользователя
}

// Определение сервиса аутентификации
service AuthService {
  // Аутентифицирует пользователя
  rpc AuthenticateUser(AuthRequest) returns (AuthResponse);
  // Регистрирует нового пользователя
  rpc RegisterUser(AuthRequest) returns (AuthResponse);
}
